python/ray/runtime_env.py

```diff
@@ -535,3 +538,49 @@ def py_container_run_options(self) -> List:
         return list(
             self._proto_runtime_env.python_runtime_env.container_runtime_env.run_options
         )
+
+    def __getstate__(self):
+        return dict(**self)
+
+    def __setstate__(self, state):
+        for k, v in state.items():
+            self[k] = v
+        self.__proto_runtime_env = None
+
+
+def get_runtime_env_info(
+    runtime_env: RuntimeEnv,
+    *,
+    is_job_runtime_env: bool = False,
+    serialize: bool = False,
+):
+    """Create runtime env info from runtime env.
+
+    In the user interface, the argument `runtime_env` contains some fields
+    which not contained in `ProtoRuntimeEnv` but in `ProtoRuntimeEnvInfo`,
+    such as `eager_install`. This function will extract those fields from
+    `RuntimeEnv` and create a new `ProtoRuntimeEnvInfo`, and serialize it.
+    """
+    proto_runtime_env_info = ProtoRuntimeEnvInfo()
+
+    proto_runtime_env_info.uris[:] = runtime_env.get_uris()
+
+    eager_install = runtime_env.get("eager_install")
+    if is_job_runtime_env or eager_install is not None:
+        if eager_install is None:
+            eager_install = True
+        elif not isinstance(eager_install, bool):
+            raise TypeError(
+                f"eager_install must be a boolean. got {type(eager_install)}"
+            )
+        proto_runtime_env_info.runtime_env_eager_install = eager_install
+
+    proto_runtime_env_info.serialized_runtime_env = runtime_env.serialize()
+
+    if not serialize:
+        return proto_runtime_env_info
+
+    return json.dumps(
```

Comment:
We don't need this json dumps/loads because we don't need to guarantee the consistency of serialized runtime env info. It is different from serialized runtime env which is used to generate a hash code.